ui/ui-components/pages/_app.tsx

```diff
@@ -10,27 +11,51 @@ import PageTransition from "../components/PageTransition";
 import StatusSubscription from "../components/StatusSubscription";
 import "../components/Icons";
 
+import { PageContext } from "../contexts/page";
+
 import { Actions } from "../utils/apiData";
-import * as eventHandlers from "../utils/eventHandlers";
+import * as eventHandlers from "../eventHandlers";
 import { getModelFromUrlOrCookie, setModelCookie } from "../utils/modelHelper";
 
-export default function GrouparooWebApp(props) {
+export interface GrouparooNextAppProps {
+  clusterName: Actions.NavigationList["clusterName"];
+  currentTeamMember: Partial<Actions.NavigationList["teamMember"]>;
+  hydrationError?: string;
+  navigation: Actions.NavigationList["navigation"];
+  navigationMode: Actions.NavigationList["navigationMode"];
+  navigationModel: Actions.NavigationList["navigationModel"];
+}
+
+export default function GrouparooNextApp(
+  props: AppProps & GrouparooNextAppProps & { err: any }
+) {
   const { Component, pageProps, err, hydrationError } = props;
 
-  const combinedProps = Object.assign({}, pageProps || {}, {
+  const combinedProps = {
```

Comment:
This `combinedProps` is still being used for legacy pages so we need to keep it. We can eliminate this When all pages are migrated to use context and imported event handlers.